piquasso/api/instruction.py

```diff
@@ -204,6 +222,75 @@ def _validate_modes(self, modes):
                 f"{self.NUMBER_OF_MODES}'."
             )
 
+    def when(self, condition: Callable) -> "Instruction":
+        """Makes the execution of the instruction conditional on a callable object.
+
+        The callable object takes a single argument - the list of measurement outcomes -
```

Comment:
Described as a tuple later on in multiple places.
```suggestion
        The callable object takes a single argument - the tuple of measurement outcomes -
```